server/service/hosts.go

```diff
@@ -192,11 +192,18 @@ func (svc *Service) ListHosts(ctx context.Context, opt fleet.HostListOptions) ([
 		return nil, err
 	}
 
-	if !opt.DisableFailingPolicies && !premiumLicense {
+	// If issues are enabled, we need to remove the critical vulnerabilities count for non-premium license.
+	// If issues are disabled, we need to explicitly set the critical vulnerabilities count to 0 for premium license.
+	if !opt.DisableIssues && !premiumLicense {
 		// Remove critical vulnerabilities count if not premium license
 		for _, host := range hosts {
 			host.HostIssues.CriticalVulnerabilitiesCount = nil
 		}
+	} else if opt.DisableIssues && premiumLicense {
```

Comment:
The reason we added `opt.DisableFailingPolicies` was for performance reasons, some customers didn't need that information on the hosts report API and it was consuming resources (taking long to execute with ~100k hosts).

Is this just to make sure to return the zero value when `opt.DisableIssues && premiumLicense`? Am guessing that now it's cheap enough so that we don't need to change ListHosts to not do the join to the pre-calculated table if `opt.DisableIssues`, right?

Reply:
Yes, this is just to return the zero value. Yes, it should be cheap enough not to need this switch.